routes/channel.js

```diff
@@ -185,15 +185,22 @@ async function retreiveLastHeartbeats(channel) {
 	)
 }
 
-function postValidatorMessages(req, res, next) {
+async function postValidatorMessages(req, res, next) {
 	if (!req.channel.spec.validators.find(v => v.id === req.session.uid)) {
 		res.sendStatus(401)
 		return
 	}
+
 	const validatorMsgCol = db.getMongo().collection('validatorMessages')
-	const { messages } = req.body
+	const channelCol = db.getMongo().collection('channels')
 
+	const { messages } = req.body
 	const startTime = Date.now()
+
+	if (req.session.uid === req.whoami && messages.find(message => message.exhausted === true)) {
```

Comment:
I think you should also check if the message is `NewState` just to be safe

This applies to both leader & follower since the leader still sends NewState to it's own sentry